### ISSUE TITLE
Average Precision Score metric

### ISSUE THREAD

**Issue by DavisVaughan** [tidymodels/yardstick#96]
As a potential improvement on PR AUC that doesn't rely on the value of `precision` in the subjective `recall = 0` case (because by design the first value of precision is removed so it doesn't matter what it was).

https://github.com/scikit-learn/scikit-learn/blob/7b136e9/sklearn/metrics/ranking.py#L226

our binary impl would be:

`sum(diff(recall) * precision[-1])`

**Comment by DavisVaughan** [comment]
Recall is well defined at the beginning (it's obviously 0 when threshold=1 or Inf), so this metric would always work

**Comment by DavisVaughan** [comment]
Implemented in #95
